app/helpers/blacklight/facets_helper_behavior.rb

```diff
@@ -136,20 +136,28 @@ def render_facet_value(facet_field, item, options ={})
   def path_for_facet(facet_field, item)
     facet_config = facet_configuration_for_field(facet_field)
     if facet_config.url_method
-      path = send(facet_config.url_method, facet_field, item)
+      send(facet_config.url_method, facet_field, item)
     else
-      path = search_action_path(add_facet_params_and_redirect(facet_field, item))
+      search_action_path(current_path.add_facet_params_and_redirect(facet_field, item))
     end
   end
 
+  def current_path
```

Comment:
Is the `UrlHelperBehavior` module a better place for this method?

Should this be memoized at all?

I was confused, at first, if `current_path` was a built-in Rails helper. I wonder if there's a better name (although I can't think of any off-hand)